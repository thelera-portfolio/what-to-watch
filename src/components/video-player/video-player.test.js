import VideoPlayer from "./video-player.jsx";
import React from "react";
import renderer from "react-test-renderer";

const Video = {
  WIDTH: 300,
  HEIGHT: 100,
  INTERVAL_IN_SEC: 1500,
  IS_MUTED: false,
};

const film = {
  preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewVideoLink: `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`,
};

it(`VideoPlayer is rendered correctly`, () => {
  const {preview, previewVideoLink} = film;

  const tree = renderer.create(
      <VideoPlayer
        duration={222}
        source={previewVideoLink}
        poster={preview}
        progress={10}
        isMuted={Video.IS_MUTED}
        isPlaying={true}
        width={Video.WIDTH}
        height={Video.HEIGHT}
        onFullScreenButtonClick={() => {}}
        onPlayButtonClick={() => {}}
      >
        <video/>
      </VideoPlayer>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
