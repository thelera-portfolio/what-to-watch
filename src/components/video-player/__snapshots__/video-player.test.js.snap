// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer is rendered correctly 1`] = `
<video>
  Sorry, your browser doesn't support embedded videos.
</video>
`;
